src/Microsoft.TemplateEngine.Edge/Installers/NuGet/NuGetInstaller.cs

```diff
@@ -401,5 +416,29 @@ private NuGetPackageInfo ReadPackageInformation(string packageLocation)
                 nuspec.GetId(),
                 nuspec.GetVersion().ToNormalizedString());
         }
+
+        // This conversion is necessary as "InstallResult" does not have the support of the NuGet types
+        private Dictionary<int, IList<string>>? ConvertVulnerabilityMetadata(IEnumerable<PackageVulnerabilityMetadata>? vulnerabilities)
+        {
+            if (vulnerabilities is null)
+            {
+                return null;
+            }
+
+            var dictionaryResult = new SortedDictionary<int, IList<string>>();
```

Comment:
not sure if dictionary is a good structure here. Imo IReadOnlyList<(int Severity, string Uri)> fits better.